pkg/cmd/server/origin/legacy_test.go

```diff
@@ -0,0 +1,26 @@
+package origin
+
+import (
+	"testing"
+
+	"github.com/openshift/origin/pkg/api/latest"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+func TestLegacyKinds(t *testing.T) {
+	for gvk := range api.Scheme.AllKnownTypes() {
+		if latest.OriginLegacyKind(gvk) {
```

Comment:
Seems like this method should be tightened to only match the whitelisted kinds.  The reflective "right package" test is useful for a test like this, but I'd expect to match "OriginKind` based on the group suffix `.openshift.io`.